From 58dec23397e3fcc4300cc03839ce5e508389abbc Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Thu, 22 Jun 2017 22:04:51 +0000
Subject: [PATCH] pthread internals: increase DEFAULT_GUARD_SIZE to 2 pages
 instead of 1 page.

This is intended to be a proactive mitigation against any bugs similar to CVE-2017-1000366.

Signed-off-by: William Pitcock <nenolod@dereferenced.org>
---
 src/internal/pthread_impl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/internal/pthread_impl.h b/src/internal/pthread_impl.h
index ae0ab1c5..ad4ea3fa 100644
--- a/src/internal/pthread_impl.h
+++ b/src/internal/pthread_impl.h
@@ -146,7 +146,7 @@ void __block_app_sigs(void *);
 void __restore_sigs(void *);
 
 #define DEFAULT_STACK_SIZE 81920
-#define DEFAULT_GUARD_SIZE 4096
+#define DEFAULT_GUARD_SIZE 8192
 
 #define __ATTRP_C11_THREAD ((void*)(uintptr_t)-1)
 
-- 
2.13.0

